esmvalcore/esgf/_search.py

```diff
@@ -97,18 +102,29 @@ def esgf_search_files(facets):
     return files
 
 
-def select_by_time(files, start_year, end_year):
-    """Select files containing data between start_year and end_year."""
+def select_by_time(files, timerange):
+    """Select files containing data between a timerange."""
     selection = []
+    start_date, end_date = _parse_period(timerange)
+
+    if start_date is None and end_date is None:
+        start_date = timerange.split('/')[0]
+        end_date = timerange.split('/')[1]
```

Comment:
Could you please move this code inside the `_parse_period` function? It appears to be duplicated everywhere the `_parse_period` function is used.